libs/ha/ha_common_libs_k8s.py

```diff
@@ -1845,3 +1846,116 @@ def dnld_obj_verify_chcksm(self, s3_test_obj, bucket, obj, download_path, upload
             return False, download_checksum
         LOGGER.info("Successfully downloaded the object and verified the checksum")
         return True, download_checksum
+
+    @staticmethod
+    def parallel_put_object(event, s3_test_obj, bkt_name: str, obj_name: str, output=None,
+                            **kwargs):
+        """
+        Function to upload object
+        Upload an object to a versioning enabled/suspended bucket and return list of dictionary of
+        uploaded versions in parallel or non parallel.
+        :param event: event to intimate thread about main thread operations
+        :param s3_test_obj: Object of the s3 test lib
+        :param bkt_name: Bucket name for calling PUT Object
+        :param obj_name: Object name for calling PUT Object
+        :param output: Output queue in which results should be put
+        :keyword file_path: File path that can be used for PUT Object call
+        :keyword count: Count for number of PUT object call
+        :keyword chk_null_version: True, if 'null' version id is expected, else False
+        :keyword background: Set to true if background function call
+        :keyword is_unversioned: Set to true if object is uploaded to an unversioned bucket
+        Can be used for setting up pre-existing objects before enabling/suspending bucket
+        versioning
+        :return: Tuple (bool, list)
+        """
+        chk_null_version = kwargs.get("chk_null_version", False)
+        file_path = kwargs.get("file_path")
+        count = kwargs.get("count", 1)
+        is_unversioned = kwargs.get("is_unversioned", False)
+        background = kwargs.get("background", False)
+        pass_put_ver = []
+        fail_put_ver = []
+        for put in range(count):
+            try:
+                put_resp = s3_test_obj.put_object(bucket_name=bkt_name, object_name=obj_name,
+                                                  file_path=file_path)
+                if is_unversioned:
+                    version_id = "null"
+                else:
+                    version_id = put_resp[1].get("VersionId", "null")
+                    if chk_null_version:
```

Comment:
Why do we need flag for this?
We always have to check version.

Reply:
> Why do we need flag for this? We always have to check version.

This is to check for un-versioned bucket, when we are expecting null version.